app/basketball/[competition]/matches/[matchid]/page.tsx

```diff
@@ -0,0 +1,144 @@
+import Custom404 from '@/components/404';
+import { asyncFetch } from '@/utils/fetch';
+import MatchStatContents from '@/components/basketball/matchStat/MatchStatContents';
+import "../../../../match.css"
+
+export default async function Page({ params, searchParams }: any) {
+    // fetch current match data
+    var match = await asyncFetch(
+        `/basketball/match/${params.matchid}`
+    );
+
+    if(!match) {
+        return (
+            <Custom404 />
+        )
+    }
+
+    // fetch all stat data of current match
+    var data = await asyncFetch(
+        `/basketball/playermatchstat?matchid=${params.matchid}&$limit=30`
+    );
+
+    var stat:BbStat[] = data.data;
+
+    var seasonid = match.season.id;
+    var teamAid = match.teama.id;
+    var teamBid = match.teamb.id;
+
+    var teamAdata : BbStat[] = [];
+    var teamBdata : BbStat[] = [];
+    var teamAtotal : BbStat = {
+        id: 0,
+        matchid: match.id,
+        playerid: 0,
+        starter: 0,
+        foul: 0,
+        points: 0,
+        assist: 0,
+        steal: 0,
+        block: 0,
+        turnover: 0,
+        offensiverebound: 0,
+        rebound: 0,
+        '3pointshit': 0,
+        '2pointshit': 0,
+        '1pointshit': 0,
+        hit: 0,
+        '3pointsshot':0,
+        '2pointsshot': 0,
+        '1pointsshot': 0,
+        shot: 0,
+        player : {
+            id: 0,
+            name: "全队",
+            email: " ",
+            weight: 0,
+            height: 0,
+            photosrc: " "
+        }
+    };
+    var teamBtotal : BbStat = {
+        id: 0,
+        matchid: match.id,
+        playerid: 0,
+        starter: 0,
+        foul: 0,
+        points: 0,
+        assist: 0,
+        steal: 0,
+        block: 0,
+        turnover: 0,
+        offensiverebound: 0,
+        rebound: 0,
+        '3pointshit': 0,
+        '2pointshit': 0,
+        '1pointshit': 0,
+        hit: 0,
+        '3pointsshot':0,
+        '2pointsshot': 0,
+        '1pointsshot': 0,
+        shot: 0,
+        player : {
+            id: 0,
+            name: "全队",
+            email: " ",
+            weight: 0,
+            height: 0,
+            photosrc: " "
+        }
+    };
+
+    for (const s of stat) {
+        // Fetch player information
+        const response = await asyncFetch(`/basketball/seasonteamplayer?seasonid=${seasonid}&playerid=${s.playerid}`);
```

Comment:
this is quite inefficient. Requesting seasonteamplayer in a loop means over 20 separate requests. 
We should call /seasonteamplayer once for each team by teamid